libp2p/NodeTable.cpp

```diff
@@ -280,10 +280,9 @@ vector<shared_ptr<NodeEntry>> NodeTable::nearestNodeEntries(NodeId _target)
 		}
 	
 	vector<shared_ptr<NodeEntry>> ret;
-	for (auto& nodes: found)
-		for (auto n: nodes.second)
-			if (ret.size() < s_bucketSize && !!n->endpoint && n->endpoint.isAllowed())
-				ret.push_back(n);
+	for (auto n: found)
```

Comment:
This changes the functionality of previous code. The code intentionally populates ret in order of the maps' key index.